tests/test_posttest.py

```diff
@@ -30,3 +32,31 @@ def test_collect_techsupport(duthost):
         duthost.fetch(src=tar_file, dest=TECHSUPPORT_SAVE_PATH, flat=True)
 
     assert True
+
+def test_restore_container_autorestart(duthost):
+    command_output = duthost.shell("show feature autorestart", module_ignore_errors=True)
+    if command_output['rc'] != 0:
+        logging.info("Feature autorestart utility not supported. Error: {}".format(command_output['stderr']))
+        logging.info("Skipping disable_container_autorestart")
+        return
+
+    state_file_name = "/tmp/autorestart_state_{}.json".format(duthost.hostname)
+    if not os.path.exists(state_file_name):
+        return
+    stored_autorestart_states = {}
+    with open(state_file_name, "r") as f:
+        stored_autorestart_states = json.load(f)
+    container_autorestart_states = duthost.get_container_autorestart_states()
```

Comment:
Why do we need to again get the container states here? Just overwriting with the stored states should do.

Reply:
The re-check here is try to reduce unnecessary update to autorestart feature. For example, the autorestart feature is disabled for database container in default, and I don't want to disable is again. Because the change of autorestart feature will cause some container reloading (like snmp) even nothing is changed.